channel-jira-server/src/main/java/com/synopsys/integration/alert/channel/jira/server/model/JiraServerGlobalConfigModel.java

```diff
@@ -0,0 +1,75 @@
+package com.synopsys.integration.alert.channel.jira.server.model;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.alert.common.rest.model.ConfigWithMetadata;
+import com.synopsys.integration.alert.common.rest.model.Obfuscated;
+
+public class JiraServerGlobalConfigModel extends ConfigWithMetadata implements Obfuscated<JiraServerGlobalConfigModel> {
+    private String url;
+    private String userName;
+    private Boolean isPasswordSet;
+    private String password;
+    private Boolean disablePluginCheck;
+
+    @Override
+    public JiraServerGlobalConfigModel obfuscate() {
+        JiraServerGlobalConfigModel jiraServerGlobalConfigModel = new JiraServerGlobalConfigModel();
+
+        jiraServerGlobalConfigModel.setId(getId());
+        jiraServerGlobalConfigModel.setName(getName());
+        jiraServerGlobalConfigModel.setLastUpdated(getLastUpdated());
+        jiraServerGlobalConfigModel.setCreatedAt(getCreatedAt());
+
+        jiraServerGlobalConfigModel.setUrl(url);
+        jiraServerGlobalConfigModel.setUserName(userName);
+        jiraServerGlobalConfigModel.setDisablePluginCheck(disablePluginCheck);
+
+        jiraServerGlobalConfigModel.setIsPasswordSet(StringUtils.isNotBlank(password));
+        jiraServerGlobalConfigModel.setPassword(null);
+
+        return jiraServerGlobalConfigModel;
+    }
+
+    public Optional<String> getUrl() {
```

Comment:
How do we feel about this optional approach for all our fields? I'm not a fan of a required field potentially being missing. I think this approach may hide other potential issues in the code base

Reply:
I understand that we did it this way for email but want to be sure we're happy with the decision

Reply:
If we removed the optionals, wouldn't places like the accessors require its own validation? Right now those are validated as part of the helper classes before the accessors are called. These models don't have constructors and are created field by field using the setters so wouldn't we run the risk of potential NPEs without the optionals?

Reply:
We will need to add tests to the action class in the future to validate that null values aren't allowed going forward.  We will need to change email, proxy, and encryption related classes in the future as well.